src/index.js

```diff
@@ -9,13 +9,20 @@ const IS_WINDOWS = platform() === 'win32';
 
 // Helper functions
 const noop = () => null;
-const matches = (key, importee) => {
+const matches = (key, importee, isRegEx) => {
   if (importee.length < key.length) {
     return false;
   }
   if (importee === key) {
     return true;
   }
+
+  if (isRegEx) {
```

Comment:
This could be `if (key instanceof RegExp)`, then the `isRegEx` flag is no longer needed.

Reply:
Also this should be moved above the other checks to the top as checking for `length` or comparing with a string does not make sense for a regular expression.

Reply:
switched to use `if (key instanceof RegExp)` and moved to the top